src/Administration/Migration/V6_7/Migration1744203319MailTemplate.php

```diff
@@ -0,0 +1,224 @@
+        $this->createMailTemplateTranslations($connection, $mailTemplateId, $languages);
+    }
+
+    private function createMailTemplateType(Connection $connection, string $mailTemplateTypeId): void
+    {
+        if ($this->mailTemplateTypeExists($connection)) {
+            return;
+        }
+
+        $connection->insert('mail_template_type', [
+            'id' => $mailTemplateTypeId,
+            'technical_name' => 'admin_sso_user_invite',
+            'available_entities' => '{}',
+            'created_at' => (new \DateTime())->format(Defaults::STORAGE_DATE_TIME_FORMAT),
+        ]);
+    }
+
+    /**
+     * @param array<string, string> $languages
+     */
+    private function createMailTemplateTypeTranslations(Connection $connection, string $mailTemplateTypeId, array $languages): void
+    {
+        $createdAt = (new \DateTime())->format(Defaults::STORAGE_DATE_TIME_FORMAT);
+
+        $translations = [];
+
+        if (\array_key_exists(self::ENGLISH_KEY, $languages)) {
+            $translations[] = [
+                'mail_template_type_id' => $mailTemplateTypeId,
+                'language_id' => $languages[self::ENGLISH_KEY],
+                'name' => 'Sso user invitation',
+                'created_at' => $createdAt,
+            ];
+        }
+
+        if (\array_key_exists(self::GERMAN_KEY, $languages)) {
+            $translations[] = [
+                'mail_template_type_id' => $mailTemplateTypeId,
+                'language_id' => $languages[self::GERMAN_KEY],
+                'name' => 'Sso Benutzer einladung',
+                'created_at' => $createdAt,
+            ];
+        }
+
+        foreach ($translations as $translation) {
+            if ($this->mailTemplateTypeTranslationsExists($connection, $mailTemplateTypeId, $translation['language_id'])) {
+                continue;
+            }
+
+            $connection->insert('mail_template_type_translation', $translation);
+        }
+    }
+
+    private function createMailTemplate(Connection $connection, string $mailTemplateId, string $mailTemplateTypeId): void
+    {
+        if ($this->mailTemplateExists($connection, $mailTemplateId)) {
+            return;
+        }
+
+        $connection->insert('mail_template', [
+            'id' => $mailTemplateId,
+            'mail_template_type_id' => $mailTemplateTypeId,
+            'system_default' => 1,
+            'created_at' => (new \DateTime())->format(Defaults::STORAGE_DATE_TIME_FORMAT),
+        ]);
+    }
+
+    /**
+     * @param array<string, string> $languages
+     */
+    private function createMailTemplateTranslations(Connection $connection, string $mailTemplateId, array $languages): void
+    {
+        $createdAt = (new \DateTime())->format(Defaults::STORAGE_DATE_TIME_FORMAT);
+
+        $translationContent = [
+            'html_en' => \file_get_contents(__DIR__ . '/assets/sso_user_invitation_mail.en-GB.html.twig'),
```

Comment:
would be nice, if we could use the Symfony FileSystem component instead

Reply:
I agree but just searching through the `src/Core/Migration` folder it looks like none of them is using that API yet and all still use `file_get_contents`. I don't want to make this PR any larger, feel free to create a ticket for it to adjust all places at the same time.